### PR TITLE
Add proposed view of integration with Analytical Platform

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Adds the proposed view of how Legal Aid Agency systems/services might integrate with the Analytical Platform.

It focuses on *team* interactions and integrations.

This diagram is a **proposed** view and will change. However, we can still merge it if the proposed state is evident.

## Why make these changes?

The Legal Aid Agency's "Data and Reporting" team is investigating the feasibility of moving to the Ministry of Justice's Analytical Platform.

I wanted to capture the *current* view of the integration.

### CODE REVIEW

**Comment by sldblog** [comment]
Any ideas on how to make the "proposed" or "draft" status more apparent? The filename? Titles?

File: diagrams/outside/analytical-platform-containers.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    tags: focus,database
+    position: '1900,200'
+  - type: Container
+    name: Subject Area Data Store
+    description: Contains data transformed from operational data to "subject areas" for reporting
+    tags: focus,database
+    position: '2700,700'
+  - type: Container
+    name: Subject Area Transformer
+    description: Transforms operational data to subject area data
```
Comment (sldblog) [inline diagrams/outside/analytical-platform-containers.yml]: 📝 Can use Apache Spark, AWS Glue, Python through jobs deployed in a Kubernetes cluster as pods

File: diagrams/outside/analytical-platform-containers.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    tags: Person,focus
+    position: '1125,650'
+  - type: Container
+    name: Operational Data Receiver
+    description: Receives and stores operational data from production applications
+    tags: focus
+    position: '1100,200'
+  - type: Container
+    name: Operational Data Store
+    description: Contains the submitted operational data
```
Comment (sldblog) [inline diagrams/outside/analytical-platform-containers.yml]: 📝 Can be AWS S3 or any data store supported by AWS that can be provisioned by the Analytical Platform.

The Analytical Platform calls this the "Landing Area" or "Land" for short. The area is owned and provisioned by the platform.

Comment (sldblog) [inline diagrams/outside/analytical-platform-containers.yml]: 📝 Can be AWS S3 or any data store supported by AWS that can be provisioned by the Analytical Platform.

The Analytical Platform calls this the "Landing Area" or "Land" for short. The area is owned and provisioned by the platform.

File: diagrams/outside/analytical-platform-containers.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  description: Hosts, processes and exposes analytical data for Ministry of Justice business teams
+  tags: focus
+  containers:
+  - type: Person
+    name: Analytical Platform Team
+    description: Owns and supports the analytical platform
+    tags: Person,focus
+    position: '1125,650'
+  - type: Container
+    name: Operational Data Receiver
```
Comment (sldblog) [inline diagrams/outside/analytical-platform-containers.yml]: 📝 There is no receiver process. The Analytical Platform expects data sources to be updated directly.
(Remove this node.)

File: diagrams/outside/analytical-platform-containers.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    description: Contains data transformed from operational data to "subject areas" for reporting
+    tags: focus,database
+    position: '2700,700'
+  - type: Container
+    name: Subject Area Transformer
+    description: Transforms operational data to subject area data
+    tags: focus
+    position: '1900,700'
+- type: Software System
+  name: LAA Business Reporting
```
Comment (sldblog) [inline diagrams/outside/analytical-platform-containers.yml]: 📝 Need to add another software system that captures the intent of RStudio and Jupyter as on-demand sandbox reporting tools as they are provided by the Analytical Platform for ad-hoc research to be used by Data Analysts.

📝 Add "Data Analysts" as a new team?

File: diagrams/outside/analytical-platform-containers.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+- source: LAA System
+  description: Makes operational data changes available
+  destination: Operational Data Receiver
+  technology: under investigation how
+- source: Operational Data Receiver
+  destination: Operational Data Store
+  technology: under investigation how
+- source: Operational Data Store
+  destination: Subject Area Transformer
+  technology: under investigation how
```
Comment (sldblog) [inline diagrams/outside/analytical-platform-containers.yml]: 📝 Each upload creates a data snapshot, versioning the data. Consequently, the transformation process has to be either very tightly controlled or compacted when personally identifiable information has to be purged.

File: diagrams/outside/analytical-platform-containers.yml (hunk 2 above)
Comment (sldblog) [reply diagrams/outside/analytical-platform-containers.yml]: In 25b32e9 I've specified S3 only, based on the Office of Public Guardian (OPG) use case.

File: diagrams/outside/analytical-platform-containers.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sldblog) [reply diagrams/outside/analytical-platform-containers.yml]: Done in 62b430d

File: diagrams/outside/analytical-platform-containers.yml (hunk 5 above)
Comment (sldblog) [reply diagrams/outside/analytical-platform-containers.yml]: I've decided not to display the above information as it's alluding to how the platform works and not that important for containers and interactions.

**Comment by ddippolito** [comment]
Interesting. A few questions:
1. Is LAA Management information team the same as Data and Reporting team? I thought Analytical platform team and MI team were part of the same team
2. There’s something I don’t understand from the diagram in relation to subject area transformer: Analytical platform team manages it but the Data and Reporting team owns the rules in it? Doesn’t that cause any issue in terms of who changes these rules?
3. In a future diagram, the two users (Analytical Platform and MI team) would be merged into one team?

**Comment by sldblog** [comment]
@ddippolito 🙇 

> 1. Is LAA Management information team the same as Data and Reporting team? I thought Analytical platform team and MI team were part of the same team

No, the LAA MI team is a non-digital functional team, working to provide LAA reports to business users.

The "Data and Reporting team" is a digital team owning some of the MI transformation processes and the business intelligence application. This diagram's "LAA Service Team" encompasses them.

> 2. There’s something I don’t understand from the diagram in relation to subject area transformer: Analytical platform team manages it but the Data and Reporting team owns the rules in it? Doesn’t that cause any issue in terms of who changes these rules?

I think the idea is: Platform team owns the infrastructure, Analyst/MI teams own the content. Perhaps I should re-label to make this explicit.

> 3. In a future diagram, the two users (Analytical Platform and MI team) would be merged into one team?

I doubt it; I think the intended split of infrastructure/content will remain the same.

**Comment by ddippolito** [comment]
@sldblog Thanks for clarifying 🎉

**Comment by sldblog** [comment]
:tophat: :bowing_man:

**Pull request merged**